### PR TITLE
Peripherals

### PR DESCRIPTION
**Pull request by martijnvandermarel**
First minimal version that works with EtherCAT.
Changed name of Temperature GES to Example_GES because it is more representative.
Includes the slave_peripherals library developed by MARCH III.

### CODE REVIEW

File: ges/Example_GES/src/main.cpp [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+// LED on DieBieSlave, for testing communication
+DigitalOut statusLed(DBS_LED);
+// Serial communication with the pc for debugging
+Serial pc(DBS_UART_USB_TX, DBS_UART_USB_RX, 128000);
+// Ethercat communication with master
+Ethercat ecat(DBS_ECAT_MOSI, DBS_ECAT_MISO, DBS_ECAT_SCK, DBS_ECAT_NCS);
+
+int main() {
+  wait(5);
```
Comment (TimBuckers) [inline ges/Example_GES/src/main.cpp]: it's better to define a variable. Other wise you have duplicated code.

File: ges/Example_GES/src/main.cpp [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  pc.printf("\f\r\nTime compiled = %s.\r\nDate = %s.\r\nBlink LED GES.", __TIME__, __DATE__);
+  statusLed = 1;
+  wait(5);
```
Comment (TimBuckers) [inline ges/Example_GES/src/main.cpp]: see comment above

File: ges/Example_GES/ESI/blinkled_conf8003.xml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        <RxPdo Fixed="0" Mandatory="1" Sm="0">
+        <Index>#x1600</Index><Name>Outputs</Name><Entry><Index>#x5</Index><SubIndex>0</SubIndex><BitLen>8</BitLen><Name>mosi_LED_command</Name><DataType>USINT</DataType><AdsInfo><AmsAddress><AmsPort>11</AmsPort></AmsAddress><IndexGroup>50593808</IndexGroup><IndexOffset>-2130706406</IndexOffset><Length>1</Length></AdsInfo></Entry></RxPdo>
+   
+<!-- ***************** PDO di input *********************************************************** -->
+   
+        <TxPdo Fixed="0" Mandatory="1" Sm="1"> 
+   		<Index>#x1a00</Index><Name>Inputs</Name><Entry><Index>#x5</Index><SubIndex>0</SubIndex><BitLen>8</BitLen><Name>miso_LED_ack</Name><DataType>USINT</DataType><AdsInfo><AmsAddress><AmsPort>11</AmsPort></AmsAddress><IndexGroup>50593808</IndexGroup><IndexOffset>-2147483618</IndexOffset><Length>1</Length></AdsInfo></Entry></TxPdo>
+          
+        <Eeprom>
+          <ByteSize>2048</ByteSize><ConfigData>80030000000000000000</ConfigData>
```
Comment (TimBuckers) [inline ges/Example_GES/ESI/blinkled_conf8003.xml]: add comment explaining the 8003

File: ges/Example_GES/src/main.cpp [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+#include "objectlist.h"
+#include "Ethercat.h"
+
+#define WAIT_TIME (2) // seconds
+#define APP_TITLE "Example GES"
+
+// LED on DieBieSlave, for testing communication
+DigitalOut statusLed(DBS_LED);
+// Serial communication with the pc for debugging
+Serial pc(DBS_UART_USB_TX, DBS_UART_USB_RX, 128000);
```
Comment (Ishadijcks) [inline ges/Example_GES/src/main.cpp]: 128000?

File: ges/Example_GES/src/main.cpp [hunk 5]
```diff
@@ -0,0 +1,41 @@
+#include <mbed.h>
+#include "DBS_pindefs.h"
+#include "objectlist.h"
+#include "Ethercat.h"
+
+#define WAIT_TIME (2) // seconds
```
Comment (Ishadijcks) [inline ges/Example_GES/src/main.cpp]: Why is this time in seconds instead of ms?

File: ges/Example_GES/src/main.cpp [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+  wait(WAIT_TIME);
+  statusLed = 0;
+
+  miso_LED_ack = 0;
+  while(1) {
+    // Update the EtherCAT buffer
+    ecat.update();
+
+    // Set led if ordered to from EtherCAT master
+    if(mosi_LED_command == 1){
```
Comment (Ishadijcks) [inline ges/Example_GES/src/main.cpp]: Refactor to this?
```
mosi_LED_command = statusLed
```

File: ges/Example_GES/src/main.cpp (hunk 5 above)
Comment (martijnvandermarel) [reply ges/Example_GES/src/main.cpp]: On a user side, seconds are more intuitive than milliseconds.

File: ges/Example_GES/src/main.cpp (hunk 6 above)
Comment (martijnvandermarel) [reply ges/Example_GES/src/main.cpp]: Refactored to statusLed = (mosi_LED_command == 1)

**Pull request merged**